### PR TITLE
Implement list decorator

### PR DESCRIPTION
**Pull request by bugagashenkj**

### CODE REVIEW

File: lib/validate.js [hunk 1]
```diff
@@ -120,6 +120,24 @@ const validateCategory = (
...
+    const opts = { ...options };
+
+    let err;
+    if (definition.domain) {
+      opts.path = `${path}${prop}`;
+      err = ms.validate('domains', definition.domain, value, opts);
+    } else if (definition.category) {
+      opts.path = `${path}${prop}.`;
+      err = ms.validate('category', definition.category, value, opts);
+    }
```
Comment (o-rumiantsev) [inline lib/validate.js]: `opts.path` is repeated in both cases

```javascript
const opts = { ...options };
opts.path = `${path}${prop}`;

let err;
if (definition.domain) {
  err = ms.validate('domains', definition.domain, value, opts);
} else if (definition.category) {
  err = ms.validate('category', definition.category, value, opts);
}
```

File: lib/validate.js [hunk 2]
```diff
@@ -158,20 +176,16 @@ const validateCategory = (
       continue;
     }
 
-    const opts = { ...options };
+    if (extractDecorator(definition) === 'List') {
+      value.forEach((ins, index) => {
```
Comment (o-rumiantsev) [inline lib/validate.js]:
```suggestion
      value.forEach((listItem, index) => {
```

File: lib/validate.js (hunk 1 above)
Comment (bugagashenkj) [reply lib/validate.js]: In the case of a domain, we have a dot at the end.

File: lib/validate.js [hunk 3]
```diff
@@ -158,20 +176,16 @@ const validateCategory = (
...
-    if (definition.domain) {
-      opts.path = `${path}${prop}`;
-      err = ms.validate('domains', definition.domain, value, opts);
-    } else if (definition.category) {
-      opts.path = `${path}${prop}.`;
-      err = ms.validate('category', definition.category, value, opts);
-    }
-    if (err) {
-      errors.push(...err.errors);
+      continue;
```
Comment (nechaido) [inline lib/validate.js]: I'd prefer not to add `continue` here, so that `definition.validate` is checked.

File: lib/decorators.js [hunk 4]
```diff
@@ -53,5 +62,6 @@ module.exports = {
     Flags: createFlags,
     // Function decorators
     Validate: fn => new Validate(fn),
+    List: createList,
```
Comment (nechaido) [inline lib/decorators.js]: `List` should be category specific decorator.

**Pull request merged**